lib/client.js

```diff
@@ -89,10 +89,14 @@ class BPClient extends Client {
       this.chain = chain;
     }
 
+    this.options.id = this.id;
+    this.options.chain = this.chain;
+
     this.setNodeClient();
     this.setWalletClient();
     this.setMultisigClient();
 
+    this.emit('set clients', this.options);
```

Comment:
Yes, I believe so, e.g. apiKey. Good call, will fix to be more deliberate. Probably just send id, chain, and maybe network?

Reply:
Those 3 sound good to me

What about the case when a dev wants to create their own `client` outside of `getClients`?  This could be useful if a dev wants to attach to multiple at the same time outside of the framework

Reply:
This utility is explicitly for interacting with the "global" client. That way any plugins that want to make sure they are using the same client as other plugins can be sure they stay up to date. In theory this could also be used with custom clients (i.e. ones not stored and interacted with via the proxy). You would just need to update the options on the singleton and then run a `setClients` to update the singleton that all other plugins are running.

Reply:
Updated, but I realize that this client doesn't necessarily have the network information since this is just talking with the proxy server. id and chain should be fine though. network can be retrieved from the state or by using `client.getInfo`